### PR TITLE
Update recognized thermal scattering materials

### PR DESCRIPTION
**Pull request by paulromano**
The next release of ENDF/B (ENDF/B-VIII.1) will have a whole bunch of new thermal scattering materials. The bulk of this PR updates our list of recognized thermal scattering materials so that these new evaluations can be processed appropriately. There was design change I had to make; namely, before we were using the MAT number on an ENDF evaluation as a way of determining what material it is. However, 1) a bunch of MAT numbers are wrongly assigned or conflict with one another and 2) the ENDF library maintainers are running out of MAT numbers.  I've changed to using the "ZSYMAM" field, which is basically an 11-character description of the material (e.g., `H(H2O)` for the light water evaluation). This should be a better long-term solution.

One other fix in this PR -- I noticed that when parsing decay sublibrary data, there was one case where we were not correctly recognizing a decay mode (when it was marked as "unknown"). That is now fixed along with some tests to ensure we get it correct.

### CODE REVIEW

**Comment by paulromano** [comment]
One additional thermal scattering-related update: NJOY2016 / NJOY21 now allows you to specify the number of moderator component ZA values when processing thermal scattering data through parameter nza on ACER card 8. I've updated our ACER template (used for `ThermalScattering.from_njoy`) to include this parameter (otherwise NJOY21 complains about it).

File: openmc/data/njoy.py [hunk 1]
```diff
@@ -445,7 +424,8 @@ def make_ace(filename, temperatures=None, acer=True, xsdir=None,
 
 def make_ace_thermal(filename, filename_thermal, temperatures=None,
                      ace='ace', xsdir=None, output_dir=None, error=0.001,
-                     iwt=2, evaluation=None, evaluation_thermal=None, **kwargs):
+                     iwt=2, evaluation=None, evaluation_thermal=None,
+                     table_name=None, zaids=None, nmix=None, **kwargs):
```
Comment (pshriwise) [inline openmc/data/njoy.py:428]: One thing that isn't clear to me is the mode of failure if a `table_name`, `zaids`, `nmix` combination is provided to the which isn't present in the thermal data file. We would get a failure from `njoy` I suppose?

Comment (pshriwise) [reply openmc/data/njoy.py:428]: Would it be possible to check that the `ThermalTuple` we're using is valid for the filenames passed before we get to parsing that data?

**Review by pshriwise — CHANGES_REQUESTED** [review]
No major comments from me here. Nice updates to the thermal data lookups for readability I'd say. Just a few probably naive questions from me to address.

While we're here, there are a fair number of unused variables in later on in `make_ace_thermal` that we could do away with to make that function shorter and easier to parse.

File: openmc/data/njoy.py (hunk 1 above)
Comment (paulromano) [reply openmc/data/njoy.py:428]: These parameters are all things that the thermal scattering evaluation _doesn't_ tell you but that you need to input in NJOY. The normal mode of operation is that we infer them from the ZSYMAM value, but in the event that a user is processing some new file that we haven't accounted for yet, these parameters give them a way to manually specify the needed parameters.

File: openmc/data/decay.py [hunk 2]
```diff
@@ -57,8 +57,11 @@ def get_decay_modes(value):
         List of successive decays, e.g. ('beta-', 'neutron')
 
     """
-    return [_DECAY_MODES[int(x)][0] for x in
-            str(value).strip('0').replace('.', '')]
+    if int(value) == 10:
```
[the remark this answers is not in the document]
Comment (paulromano) [reply openmc/data/decay.py:60]: Just added a comment

**Comment by paulromano** [comment]
@pshriwise although your IDE may tell you that there are bunch of unused variables, they are actually used in the NJOY templates, where we substitute them with `.format(**locals())`

**Comment by pshriwise** [comment]
> @pshriwise although your IDE may tell you that there are bunch of unused variables, they are actually used in the NJOY templates, where we substitute them with `.format(**locals())`

Ah, gotcha. Wasn't looking for that. Thanks!

**Review by pshriwise — APPROVED** [review]
Thanks @paulromano!

**Pull request merged**